test/functional/api/v2/compute_resources_controller_test.rb

```diff
@@ -289,4 +289,18 @@ def teardown
     cr.reload
     assert_nil cr.attrs[:setpw]
   end
+
+  test "should not update display_type for non-Libvirt compute resource" do
+    cr = compute_resources(:openstack)
+    put :update, { :id => cr.id, :compute_resource => { :display_type => 'SPICE' } }
```

Comment:
This test is silently failing, as it's missing an `assert_response :success` (ditto the ones above, actually).  The difference between it and set_console_password is that the latter has a no-op method in ComputeResource, while display_type doesn't so it'll fail with an unknown method/attr error.